### PR TITLE
Tag deployed lambda functions

### PR DESCRIPTION
**Pull request by jamesls**
Makes it possible for customers to easily see
which lambda functions are used for chalice.
Also opens up the possibility of us not having
to rely on the local deployed.json.  That would
be future work though.  This commit
only adds support for tagging the functions.

The format uses colon separated values because
Lambda does not allow CSV values for the value
of a tag.

I also bumped the min version of botocore to the
version where tagging support was added to lambda.

### CODE REVIEW

File: chalice/awsclient.py [hunk 1]
```diff
@@ -64,8 +64,9 @@ def get_function_configuration(self, name):
         return response
 
     def create_function(self, function_name, role_arn, zip_contents,
-                        environment_variables=None, runtime='python2.7'):
-        # type: (str, str, str, _ENV_VAR, str) -> str
+                        environment_variables=None, runtime='python2.7',
+                        tags=None):
+        # type: (str, str, str, _STR_MAP, str, Optional[Dict[str, str]]) -> str
```
Comment (kyleknap) [inline chalice/awsclient.py]: Why was ``_STR_MAP`` not used for tags type annotation? ``_STR_MAP`` is the same as ``Optional[Dict[str,str]]`` right?

**Comment by kyleknap** [comment]
Oh also do we need to update the SAM template generator for this as well?

**Comment by jamesls** [comment]
@kyleknap Looks like I didn't push the SAM changes.  I have it coded locally, I'll get this updated shortly.

**Comment by jamesls** [comment]
@stealthycoin Yeah that will be part of the PR that adds support for user tagging.

**Comment by jamesls** [comment]
@kyleknap Feedback incorporated should be good to look at again.

**Pull request merged**